### PR TITLE
Model Card Metadata documentation update

### PR DESCRIPTION
**Pull request by davanstrien**
- Restructure page 
- Add section on metadata ui
- add some more FAQs

TODO: 
- [x] Document `not-for-all-audiences`
- [x] Document custom licences

### CODE REVIEW

**Comment by davanstrien** [comment]
cc @meg-huggingface @EziOzoani

File: docs/hub/model-cards.md [hunk 1]
```diff
@@ -21,7 +21,41 @@ The model card template is available [here](https://github.com/huggingface/huggi
...
+### Adding metadata to your model card
+
+There are a few different ways to add metadata to your model card including:
+- Using the metadata UI
+- Directly editing the YAML section of the `README.md` file
+- Via the [`huggingface_hub`](https://huggingface.co/docs/huggingface_hub) Python library, see the [docs](https://huggingface.co/docs/huggingface_hub/guides/model-cards#update-metadata) for more details.
+
+Many libraries with [Hub integration](./models-libraries.md) will automatically add metadata to the model card when you upload a model. 
+
+### Using the metadata UI
```
Comment (osanseviero) [inline docs/hub/model-cards.md:44]: Should this be a h4 rather than a h3?

File: docs/hub/model-cards.md [hunk 2]
```diff
@@ -21,7 +21,41 @@ The model card template is available [here](https://github.com/huggingface/huggi
...
+
+You can add metadata to your model card using the metadata UI. To access the metadata UI, go to the model page and click on the `Edit model card` button in the top right corner of the model card. This will open an editor showing the model card `README.md` file, as well as a UI for editing the metadata.
+
+
+<div class="flex justify-center">
+<img class="block dark:hidden" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/548b926bfb572ec93b798548a64755e6e68a495a/hub/metadata-ui-editor.png"/>
```
Comment (osanseviero) [inline docs/hub/model-cards.md:50]: To update before merge

File: docs/hub/model-cards.md [hunk 3]
```diff
@@ -89,8 +117,21 @@ This metadata will be used to display the base model on the model page. Users ca
 <img class="hidden dark:block" src="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/hub/base-model-ui-dark.png"/>
 </div>
 
+### Specifying a dataset
```
Comment (osanseviero) [inline docs/hub/model-cards.md:120]: I would link this from https://moon-ci-docs.huggingface.co/docs/hub/pr_1116/en/models-faq#how-can-i-see-what-dataset-was-used-to-train-the-model

File: docs/hub/model-cards.md [hunk 4]
```diff
@@ -89,8 +117,21 @@ This metadata will be used to display the base model on the model page. Users ca
...
+datasets:
+- imdb
+- HuggingFaceH4/no_robots
+```
+
+### Specifying a task (`pipeline_tag`)
+
+You can specify the `pipeline_tag` in the model card metadata. The `pipeline_tag` indicates the type of task the model is intended for. This tag will be displayed on the model page and users can filter models on the Hub by task. This tag is also used to determine which [widget](./models-widgets.md#enabling-a-widget) to use for the model. 
+
+For `transformers` models, the pipeline tag is automatically inferred from the model's `config.json` file but you can override it in the model card metadata if required. Editing this field in the metadata UI will ensure that the pipeline tag is valid. Some other libraries with Hub integration will also automatically add the pipeline tag to the model card metadata.
```
Comment (osanseviero) [inline docs/hub/model-cards.md:134]: Some of this is duplicated with https://moon-ci-docs.huggingface.co/docs/hub/pr_1003/en/models-widgets#enabling-a-widget. Let's unify these. Probably we can move some of that over here, including the gist

**Review by osanseviero — COMMENTED** [review]
Thank you! :fire:

File: docs/hub/model-cards.md [hunk 5]
```diff
@@ -21,7 +21,41 @@ The model card template is available [here](https://github.com/huggingface/huggi
...
+
+
+### Adding metadata to your model card
+
+There are a few different ways to add metadata to your model card including:
+- Using the metadata UI
+- Directly editing the YAML section of the `README.md` file
+- Via the [`huggingface_hub`](https://huggingface.co/docs/huggingface_hub) Python library, see the [docs](https://huggingface.co/docs/huggingface_hub/guides/model-cards#update-metadata) for more details.
+
+Many libraries with [Hub integration](./models-libraries.md) will automatically add metadata to the model card when you upload a model.
```
Comment (osanseviero) [inline docs/hub/model-cards.md:42]:
```suggestion
Many libraries with [Hub integration](./models-libraries) will automatically add metadata to the model card when you upload a model. 
```

File: docs/hub/model-cards.md [hunk 6]
```diff
@@ -40,23 +74,17 @@ datasets:
...
 
-The metadata that you add to the model card enables certain interactions on the Hub. For example:
-* Allow users to filter and discover models at https://huggingface.co/models.
-* If you choose a license using the keywords listed in the right column of [this table](./repositories-licenses), the license will be displayed on the model page.
-* Adding datasets to the metadata will add a message reading `Datasets used to train:` to your model card and link the relevant datasets, if they're available on the Hub.
-
-Dataset, metric, and language identifiers are those listed on the [Datasets](https://huggingface.co/datasets), [Metrics](https://huggingface.co/metrics) and [Languages](https://huggingface.co/languages) pages and in the [`datasets`](https://github.com/huggingface/datasets) repository.
-
-See the detailed model card metadata specification [here](https://github.com/huggingface/hub-docs/blob/main/modelcard.md?plain=1).
+You can find the detailed model card metadata specification [here](https://github.com/huggingface/hub-docs/blob/main/modelcard.md?plain=1).
```
Comment (osanseviero) [inline docs/hub/model-cards.md:81]: Let's maybe open this in a new tab

File: docs/hub/model-cards.md [hunk 7]
```diff
@@ -89,8 +117,38 @@ This metadata will be used to display the base model on the model page. Users ca
...
 
+```yaml
+datasets:
+- imdb
+- HuggingFaceH4/no_robots
+```
+
+### Specifying a task (`pipeline_tag`)
+
+You can specify the `pipeline_tag` in the model card metadata. The `pipeline_tag` indicates the type of task the model is intended for. This tag will be displayed on the model page and users can filter models on the Hub by task. This tag is also used to determine which [widget](./models-widgets.md#enabling-a-widget) to use for the model.
```
Comment (osanseviero) [inline docs/hub/model-cards.md:132]:
```suggestion
You can specify the `pipeline_tag` in the model card metadata. The `pipeline_tag` indicates the type of task the model is intended for. This tag will be displayed on the model page and users can filter models on the Hub by task. This tag is also used to determine which [widget](./models-widgets.md#enabling-a-widget) to use for the model and which APIs to use under the hood.
```

**Review by osanseviero — APPROVED** [review]
Looks great! :fire:

**Pull request merged**